app/javascript/components/widgets/numbers/falling-blocks.jsx

```diff
@@ -43,10 +51,23 @@ class Scene extends React.Component {
       const loadedWorld = this.res.resurrect(world);
       Engine.merge(this.engine, { world: loadedWorld });
     }
+    Composite.allBodies(this.engine.world)
+      .filter(body => body.isStatic && body.render.sprite.texture)
+      .map(b => this.addToOverlay(b));
+    Events.on(this.engine, 'beforeUpdate', b => this.prepareOverlay(b));
     this.addBlocks();
   }
 
   componentWillUnmount() {
+    const allBodies = Composite.allBodies(this.engine.world);
+    allBodies
+      .filter(body => body.onOverlay)
+      .map(body => {
+        const b = body;
+        b.onOverlay = false;
```

Comment:
`is<something>`, `has<something>`, `should<something>` are better hints at bools. `on<something` suggests a callback.